src/control/security/config_test.go

```diff
@@ -0,0 +1,209 @@
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package security
+
+import (
+	"os"
+	"strings"
+	"bytes"
+	"testing"
+)
+
+func InsecureTC() *TransportConfig{
+	return &TransportConfig{
+		AllowInsecure: true,
+	}
+}
+func BadTC() *TransportConfig{
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath: "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/bad.crt",
+			PrivateKeyPath: "testdata/certs/bad.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+func ServerTC() *TransportConfig {
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath: "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/server.crt",
+			PrivateKeyPath: "testdata/certs/server.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+func AgentTC() *TransportConfig {
+	return &TransportConfig{
+		AllowInsecure: false,
+		CertificateConfig: CertificateConfig{
+			CARootPath: "testdata/certs/daosCA.crt",
+			CertificatePath: "testdata/certs/agent.crt",
+			PrivateKeyPath: "testdata/certs/agent.key",
+			tlsKeypair:      nil,
+			caPool:          nil,
+		},
+	}
+}
+func SetupTCFilePerms(t *testing.T, conf *TransportConfig) {
+	if err := os.Chmod(conf.CARootPath, SafeCertPerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(conf.CertificatePath, SafeCertPerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(conf.PrivateKeyPath, SafeKeyPerm); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func ValidateInsecure (t *testing.T, c *TransportConfig, err error){
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.tlsKeypair != nil || c.caPool != nil {
+		t.Fatal("insecure config loaded certs")
+	}
+}
+func ValidateGood (t *testing.T, c *TransportConfig, err error){
+	if err != nil {
+		t.Fatal(err)
+	}
+	if c.tlsKeypair == nil || c.caPool == nil {
+		t.Fatal("certs did not load yet returned no error")
+	}
+}
+func ValidateBad (t *testing.T, c *TransportConfig, err error){
+	if err == nil {
+		t.Fatal(err)
+	}
+}
+func ValidateNil (t *testing.T, c *TransportConfig, err error){
+	if err != nil &&
+		strings.Compare(err.Error(), "nil TransportConfig") != 0 {
+		t.Fatal(err)
+	}
+}
+func TestPreLoadCertData(t *testing.T) {
+	insecureTC := InsecureTC()
+	serverTC := ServerTC()
+	badTC := BadTC()
+
+	// Setup permissions for tests below.
+	SetupTCFilePerms(t, serverTC)
+	SetupTCFilePerms(t, badTC)
+
+	testCases := []struct {
+		testname string
+		config *TransportConfig
+		Validate func(t *testing.T, c *TransportConfig, err error)
+	}{
+		{"InsecureTC", insecureTC, ValidateInsecure},
+		{"GoodTC", serverTC, ValidateGood},
+		{"BadTC", badTC, ValidateBad},
+		{"NilTC", nil, ValidateNil},
```

Comment:
This is fine, but feels like overkill given how little logic is shared, and no params are shared. Just a suggestion, but I'd probably just split these into separate test functions, rather than trying to use the table testing format. May even be fewer lines of code.

Reply:
I just didn't want to have to set the correct permissions on the files multiple times. This way its done once in this function and used multiple times.

Reply:
The file permission setup seems like it's easily shared, though, since it's split out into its own function. I don't feel that strongly about it, it just seems a little overcomplicated.